ui/ozone/platform/starboard/gl_ozone_egl_starboard.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ui/ozone/platform/starboard/gl_ozone_egl_starboard.h"
+
+#include <memory>
+
+#include "starboard/egl.h"
+#include "starboard/window.h"
+#include "ui/gfx/vsync_provider.h"
+
+namespace {
+// TODO(b/371272304): Currently targeting a fixed 60Hz. Make this dynamic based
+// on device capabilities.
+base::TimeDelta GetVSyncInterval() {
+  return base::Seconds(1) / 60;
+}
+
+}  // namespace
+
+namespace ui {
+
+GLOzoneEGLStarboard::GLOzoneEGLStarboard() = default;
+
+GLOzoneEGLStarboard::~GLOzoneEGLStarboard() {
+  // TODO(b/371272304): Terminate the display and release any resources.
```

Comment:
Neat, I didn't know about custom deleters.

Looking at existing usage of SbWindow it seems like it is generally designed to be used without a pointer (all of the functions that have a SbWindow param are by value). Would we still want to use a unique_ptr and custom deleter in that case, or just call SbWindowDestroy it in the destructor here?

Reply:
For now I've implemented without unique_ptr/deleter to follow other usage I'm seeing.

Reply:
I'd advocate for the unique_ptr<> because in general it's best to not have explicit destruction. That in itself justifies the extra cost of having a heap-allocated objec(as opposed to stack objects).

Reply:
I'm running into issues with this implementation. The `unique_ptr` expects dynamic allocation, and with `SbWindowCreate` we are dynamically allocating `SbWindowPrivate`, but not `SbWindow` (because it's treated as an object). Using `unique_ptr<SbWindowPrivate>` *would* work since we can cast SbWindow->SbWindowPrivate*, but we don't have the definition of `SbWindowPrivate` outside of Starboard so it fails due to an undefined type. I also tried `unique_ptr<void>` and learned that the compiler rejects that.

I spent some time working with @y4vor to look at this and we could get it working with a bunch of `reinterpret_cast`s every time we set or accessed it, but we decided that was not clean or preferred.

Reply:
I see, well, in the interest of time let's not block this CL on this specific lifetime mgmt.